transaction/std/src/memo_builder/gift_code_cancellation_memo_builder.rs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Defines the Memo Builder for the gift code cancellation memo (0x0202)
+//! specified in MCIP #32
+
+use super::{
+    memo::{GiftCodeCancellationMemo, UnusedMemo},
+    MemoBuilder, ReservedDestination,
+};
+use mc_account_keys::PublicAddress;
+use mc_transaction_core::{Amount, MemoContext, MemoPayload, NewMemoError};
+
+/// There are three possible gift code memo types specified in MCIP #32
+/// | Memo type bytes | Name                        |
+/// | -----------     | -----------                 |
+/// |    0x0002       | Gift Code Sender Memo       |
+/// |    0x0201       | Gift Code Funding Memo      |
+/// | -->0x0202<--    | Gift Code Cancellation Memo |
+/// This memo builder builds a gift code cancellation memo (0x0202). Gift code
+/// cancellation is defined as the sender sending the gift code TxOut at the
+/// gift code subaddress back to their default address prior to it being spent
+/// by the receiver. When that happens a change TxOut is sent to the gift
+/// code sender's change subaddress with a gift code cancellation memo that
+/// stores the index of the TxOut originally sent to the gift code subaddress
+/// when the gift code was funded.
```

Comment:
The MCIP left it unclear. Originally I was thinking it would be more consistent to have it go back to the funding address, but especially point number 2. I agree we should send it to the change address.